### PR TITLE
1password: 1.0.0 -> 1.1.1

### PR DESCRIPTION
**Pull request by IvarWithoutBones**
###### Motivation for this change 
Release notes are available [here](https://app-updates.agilebits.com/product_history/CLI).

This also removes some duplicate code in the `src` fetching, what I've done is a bit cleaner IMO.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/1password/default.nix [hunk 1]
```diff
@@ -2,26 +2,22 @@
...
+      url = {
+        "i686-linux" = "https://cache.agilebits.com/dist/1P/op/pkg/v${version}/op_linux_386_v${version}.zip";
+        "x86_64-linux" = "https://cache.agilebits.com/dist/1P/op/pkg/v${version}/op_linux_amd64_v${version}.zip";
+      }.${stdenv.hostPlatform.system};
+      sha256 = {
+        "i686-linux" = "1andl3ripkcg4jhwdkd4b39c9aaxqpx9wzq21pysn6rlyy4hfcb0";
+        "x86_64-linux" = "0qj5v8psqyp0sra0pvzkwjpm28kx3bgg36y37wklb6zl2ngpxm5g";
+      }.${stdenv.hostPlatform.system};
+      stripRoot = false;
+    } else fetchurl {
```
Comment (ryantm) [inline pkgs/applications/misc/1password/default.nix:17]: I think it was better before how it defaulted to throwing if it couldn't match the architecture. What's the reason for that change?

**Comment by ryantm** [comment]
@GrahamcOfBorg build 1password

**Review by ryantm — CHANGES_REQUESTED** [review]
Please squash the commits and rename the commit message to use the attrpath `_1password`.

File: pkgs/applications/misc/1password/default.nix (hunk 1 above)
Comment (IvarWithoutBones) [reply pkgs/applications/misc/1password/default.nix:17]: I thought it looked a bit off, as nix would throw an error before that would run. The specified `meta.platforms` include all platforms which are supported already, and I'm fairly sure nix would error out if the system doesn't match one of those.

**Comment by ryantm** [comment]
@GrahamcOfBorg build _1password

**Pull request merged**